lab_5_scrapper/scrapper.py

```diff
@@ -120,45 +263,141 @@ def __init__(self, full_url: str, article_id: int, config: Config) -> None:
+            "марта": "March",
+            "апреля": "April",
+            "мая": "May",
+            "июня": "June",
+            "июля": "July",
+            "августа": "August",
+            "сентября": "September",
+            "октября": "October",
+            "ноября": "November",
+            "декабря": "December"
+        }
+        date_str = date_str.replace(date_str.split()[1], months_dict[date_str.split()[1]])
+        date_obj = str(datetime.datetime.strptime(date_str, '%d %B %Y - %H:%M'))
+        return datetime.datetime.strptime(date_obj, '%Y-%m-%d %H:%M:%S')
 
     def parse(self) -> Union[Article, bool, list]:
         """
         Parses each article
         """
-        pass
+        response = make_request(self.full_url, self.config)
+        article_soup = BeautifulSoup(response.text, 'lxml')
+        self._fill_article_with_text(article_soup)
+        self._fill_article_with_meta_information(article_soup)
+        return self.article
 
 
 def prepare_environment(base_path: Union[Path, str]) -> None:
     """
     Creates ASSETS_PATH folder if no created and removes existing folder
     """
-    pass
+    if base_path.exists():
+        shutil.rmtree(base_path)
+    base_path.mkdir(parents=True)
+
+
+class CrawlerRecursive(Crawler):
+    """
+    Recursive Crawler implementation
+    """
+
+    def __init__(self, config: Config) -> None:
+        """
+        Initializes an instance of the Recursive Crawler class
+        """
+        super().__init__(config)
+        self.start_url = self.get_search_urls()[0]
+        self.visited_urls = set()
+
+    def find_articles(self) -> None:
+        """
+        Finds articles recursively starting from the given URL
+        """
+        if len(self.visited_urls) >= self._config.get_num_articles():
+            return
+        response = make_request(self.start_url, self._config)
+        article_bs = BeautifulSoup(response.text, 'lxml')
+        for paragraph in article_bs.find_all('a', class_='widget-view-small__head'):
+            article_url = self._extract_url(paragraph)
+            if article_url is None or 'http' not in article_url:
+                continue
+            if article_url in self.visited_urls:
+                return
+            self.visited_urls.add(article_url)
+            self.urls.append(article_url)
+            if len(self.visited_urls) < self._config.get_num_articles():
+                self.start_url = article_url
+                self.find_articles()
 
 
 def main() -> None:
     """
     Entrypoint for scrapper module
     """
-    pass
+    config = Config(CRAWLER_CONFIG_PATH)
+    prepare_environment(ASSETS_PATH)
+    crawler = Crawler(config)
+    crawler.find_articles()
+    for ind, url in enumerate(crawler.urls, 1):
+        parser = HTMLParser(url, ind, config)
+        article = parser.parse()
+        if isinstance(article, Article):
+            to_raw(article)
+            to_meta(article)
+
+    config = Config(CRAWLER_CONFIG_PATH)
```

Comment:
1. split main with standard crawler and recursive
1. leave main "main" with old name, rename new "main" to "main_recursive"